### PR TITLE
refactor(nix): add shell completion and avoid impure

### PR DESCRIPTION
**Pull request by XYenon**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -2,16 +2,30 @@
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
     flake-utils.url = "github:numtide/flake-utils";
-    devenv = {
-      url = "github:cachix/devenv";
-      inputs.nixpkgs.follows = "nixpkgs";
+    rust-overlay = {
```
Comment (teto) [inline flake.nix:5]: we need the rust overlay because yazi relies on a more recent rust than what is nixpkgs ?

**Review by teto — APPROVED** [review]
thanks a lot, that worked .

File: flake.nix (hunk 1 above)
Comment (XYenon) [reply flake.nix:5]: `rust-src` is needed to make rust-analyzer works, is there any way to get it from nixpkgs?

Comment (teto) [reply flake.nix:5]: rust-analyzer seems to be in nixpkgs but I dont have much experience with the whole process so I trust you :)

**Comment by sxyazi** [comment]
Thanks, let me merge it now!

**Pull request merged**